group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -739,6 +747,12 @@ public Optional<OffsetExpirationCondition> offsetExpirationCondition() {
         return Optional.of(new OffsetExpirationConditionImpl(offsetAndMetadata -> offsetAndMetadata.commitTimestampMs));
     }
 
+    @Override
+    public boolean isInStatesCaseInsensitive(List<String> statesFilter, long committedOffset) {
+        Set<String> caseInsensitiveFilterSet = statesFilter.stream().map(String::toLowerCase).map(String::trim).collect(Collectors.toSet());
```

Comment:
It is a tad annoying to call `toLowerCase` here because we have to do it for every group. Would it be better to keep this part on the calling side?

Reply:
Done.